(Enhancement) Adds features to fio image

mittachaitu opened the pull request:
**What this PR does / why we need it**:
  
This PR adds below feature to the existing fio job
```
./fio_runner.sh --help
Usage :  fio_runner.sh --template
              fio_runner.sh --size
              fio_runner.sh --duration
              fio_runner.sh --help	
	      fio_runner.sh --available_space
	      fio_runner.sh --miscellaneous
	      fio_runner.sh --read-only
	      fio_runner.sh --write-only

-h|--help    		Display this help and exit  
--template   		Select the fio template to run 
--size	     		Provide the data sample size (in Bytes as recomonded)
--duration   		Duration (in sec)
--available_space 	Get space left at mount point
--miscellaneous		Perform writes and reads on available space on given mount point
			Ex: ./fio_runner.sh --available_space /datadir --miscellaneous /datadir(mount point)
--read-only		Perform Read on the filepath which is specified(Requires file path)
--write-only		Perform Writes using fio profile(Optional argument to perform writes)

Example: ./fio_runner.sh --template file/basic-rw --size 1024m --duration 120
```


Signed-off-by: mittachaitu <sai.chaithanya@mayadata.io>

<!--  Thanks for sending a pull request!  Here are some tips for you -->

ksatchit commented:
@mittachaitu , it would be good to post results of existing fio perf runner litmus job (being used pretty regularly by our users: https://github.com/openebs/litmus/tree/master/apps/fio/tests/performance)

mittachaitu commented:
@ksatchit logs of (https://github.com/openebs/litmus/tree/master/apps/fio/tests/performance) litmus job:
```
kubectl logs -f  litmus-fio-nf9jk-5q9bb -n litmus
No config file found; using defaults
/etc/ansible/hosts did not meet host_list requirements, check plugin documentation if this is unexpected
/etc/ansible/hosts did not meet script requirements, check plugin documentation if this is unexpected
[DEPRECATION WARNING]: Included file '/fio/tests/performance/test_cleanup.yaml'
 not found, however since this include is not explicitly marked as 'static: 
yes', we will try and include it dynamically later. In the future, this will be
 an error unless 'static: no' is used on the include task. If you do not want 
missing includes to be considered dynamic, use 'static: yes' on the include or 
set the global ansible.cfg options to make all includes static for tasks and/or
 handlers. This feature will be removed in version 2.12. Deprecation warnings 
can be disabled by setting deprecation_warnings=False in ansible.cfg.

PLAY [localhost] ***************************************************************
2019-01-18T04:15:06.256878 (delta: 0.034414)         elapsed: 0.034414 ******** 
=============================================================================== 

TASK [Gathering Facts] *********************************************************
2019-01-18T04:15:06.264731 (delta: 0.007829)         elapsed: 0.042267 ******** 
ok: [127.0.0.1]

TASK [Check whether the provider storageclass is applied] **********************
2019-01-18T04:15:09.479339 (delta: 3.214592)         elapsed: 3.256875 ******** 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get sc openebs-cstor-sparse", "delta": "0:00:00.570790", "end": "2019-01-18 04:15:10.257856", "rc": 0, "start": "2019-01-18 04:15:09.687066", "stderr": "", "stderr_lines": [], "stdout": "NAME                   PROVISIONER                    AGE\nopenebs-cstor-sparse   openebs.io/provisioner-iscsi   52m", "stdout_lines": ["NAME                   PROVISIONER                    AGE", "openebs-cstor-sparse   openebs.io/provisioner-iscsi   52m"]}

TASK [Replace the app node placeholder with perf-intensive node] ***************
2019-01-18T04:15:10.291523 (delta: 0.812158)         elapsed: 4.069059 ******** 
ok: [127.0.0.1] => {"changed": false, "msg": ""}

TASK [Replace the pvc placeholder with test param] *****************************
2019-01-18T04:15:10.511436 (delta: 0.219888)         elapsed: 4.288972 ******** 
changed: [127.0.0.1] => {"changed": true, "msg": "2 replacements made"}

TASK [Replace the storageclass placeholder with provider] **********************
2019-01-18T04:15:10.641113 (delta: 0.129652)         elapsed: 4.418649 ******** 
changed: [127.0.0.1] => {"changed": true, "msg": "1 replacements made"}

TASK [Replace the default fio profile with user-defined profile] ***************
2019-01-18T04:15:10.775332 (delta: 0.134192)         elapsed: 4.552868 ******** 
changed: [127.0.0.1] => {"changed": true, "msg": "1 replacements made"}

TASK [Replace the data sample size with user-defined size] *********************
2019-01-18T04:15:10.911662 (delta: 0.136301)         elapsed: 4.689198 ******** 
changed: [127.0.0.1] => {"changed": true, "msg": "1 replacements made"}

TASK [Replace the default I/O test duration with user-defined period] **********
2019-01-18T04:15:11.044980 (delta: 0.133294)         elapsed: 4.822516 ******** 
ok: [127.0.0.1] => {"changed": false, "msg": ""}

TASK [Deploy fio test job] *****************************************************
2019-01-18T04:15:11.177180 (delta: 0.132174)         elapsed: 4.954716 ******** 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl apply -f fio.yml -n litmus", "delta": "0:00:00.483073", "end": "2019-01-18 04:15:11.757432", "rc": 0, "start": "2019-01-18 04:15:11.274359", "stderr": "", "stderr_lines": [], "stdout": "job.batch/fio created\npersistentvolumeclaim/fio-benchmark created", "stdout_lines": ["job.batch/fio created", "persistentvolumeclaim/fio-benchmark created"]}

TASK [Confirm fio pod status is running] ***************************************
2019-01-18T04:15:11.794145 (delta: 0.616937)         elapsed: 5.571681 ******** 
FAILED - RETRYING: Confirm fio pod status is running (15 retries left).
changed: [127.0.0.1] => {"attempts": 2, "changed": true, "cmd": "kubectl get pods -l name=fio -n litmus --no-headers", "delta": "0:00:00.397108", "end": "2019-01-18 04:17:12.846161", "rc": 0, "start": "2019-01-18 04:17:12.449053", "stderr": "", "stderr_lines": [], "stdout": "fio-q5dtm   1/1   Running   0     2m", "stdout_lines": ["fio-q5dtm   1/1   Running   0     2m"]}

TASK [Obtain name of fio pod] **************************************************
2019-01-18T04:17:12.881772 (delta: 121.087596)         elapsed: 126.659308 **** 
ok: [127.0.0.1] => {"ansible_facts": {"fio_pod_name": "fio-q5dtm"}, "changed": false}

TASK [Wait for fio pod to proceed with workload] *******************************
2019-01-18T04:17:12.929862 (delta: 0.048066)         elapsed: 126.707398 ****** 
ok: [127.0.0.1] => {"changed": false, "elapsed": 60, "path": null, "port": null, "search_regex": null, "state": "started"}

TASK [Re-Check fio pod status] *************************************************
2019-01-18T04:18:13.232034 (delta: 60.302149)         elapsed: 187.00957 ****** 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pod fio-q5dtm -n litmus --no-headers -o custom-columns=:status.phase", "delta": "0:00:00.385350", "end": "2019-01-18 04:18:13.723198", "failed_when_result": false, "rc": 0, "start": "2019-01-18 04:18:13.337848", "stderr": "", "stderr_lines": [], "stdout": "Running", "stdout_lines": ["Running"]}

TASK [Confirm fio job completion status (poll & wait for upperbound of 60 min)] ***
2019-01-18T04:18:13.760568 (delta: 0.528509)         elapsed: 187.538104 ****** 
FAILED - RETRYING: Confirm fio job completion status (poll & wait for upperbound of 60 min) (30 retries left).
FAILED - RETRYING: Confirm fio job completion status (poll & wait for upperbound of 60 min) (29 retries left).
FAILED - RETRYING: Confirm fio job completion status (poll & wait for upperbound of 60 min) (28 retries left).
FAILED - RETRYING: Confirm fio job completion status (poll & wait for upperbound of 60 min) (27 retries left).
changed: [127.0.0.1] => {"attempts": 5, "changed": true, "cmd": "kubectl get pod fio-q5dtm -n litmus --no-headers -o custom-columns=:status.phase", "delta": "0:00:00.560966", "end": "2019-01-18 04:26:16.675440", "rc": 0, "start": "2019-01-18 04:26:16.114474", "stderr": "", "stderr_lines": [], "stdout": "Succeeded", "stdout_lines": ["Succeeded"]}

TASK [Verify the fio logs to check if run is complete w/o errors] **************
2019-01-18T04:26:16.712373 (delta: 482.951781)         elapsed: 670.489909 **** 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl logs fio-q5dtm -n litmus | grep -i error | cut -d \":\" -f 2 | sort | uniq", "delta": "0:00:00.407113", "end": "2019-01-18 04:26:17.220192", "failed_when_result": false, "rc": 0, "start": "2019-01-18 04:26:16.813079", "stderr": "", "stderr_lines": [], "stdout": " 0,", "stdout_lines": [" 0,"]}

TASK [set_fact] ****************************************************************
2019-01-18T04:26:17.257505 (delta: 0.545104)         elapsed: 671.035041 ****** 
ok: [127.0.0.1] => {"ansible_facts": {"flag": "Pass"}, "changed": false}

TASK [include] *****************************************************************
2019-01-18T04:26:17.303671 (delta: 0.04614)         elapsed: 671.081207 ******* 
fatal: [127.0.0.1]: FAILED! => {"reason": "Unable to retrieve file contents\nCould not find or access '/fio/tests/performance/test_cleanup.yaml' on the Ansible Controller.\nIf you are using a module and expect the file to exist on the remote, see the remote_src option"}
	to retry, use: --limit @/fio/tests/performance/test.retry

PLAY RECAP *********************************************************************
127.0.0.1                  : ok=16   changed=10   unreachable=0    failed=1   

2019-01-18T04:26:17.332056 (delta: 0.028353)         elapsed: 671.109592 ****** 
=============================================================================== 
```
fio-readwrite job logs
```
kubectl logs -f fio-q5dtm -n litmus

Running standard-ssd test with size=2048m, runtime=60... Wait for results !!

{
  "fio version" : "fio-2.2.10",
  "timestamp" : 1547785475,
  "time" : "Fri Jan 18 04:24:35 2019",
  "jobs" : [
    {
      "jobname" : "seq-read",
      "groupid" : 0,
      "error" : 0,
      "eta" : 0,
      "elapsed" : 242,
      "read" : {
        "io_bytes" : 166232,
        "bw" : 2770,
        "iops" : 692.62,
        "runtime" : 60001,
        "total_ios" : 41558,
        "short_ios" : 0,
        "drop_ios" : 0,
        "slat" : {
          "min" : 7,
          "max" : 101,
          "mean" : 14.39,
          "stddev" : 3.30
        },
        "clat" : {
          "min" : 1056,
          "max" : 62050,
          "mean" : 1425.94,
          "stddev" : 730.54,
          "percentile" : {
            "1.000000" : 1176,
            "5.000000" : 1208,
            "10.000000" : 1224,
            "20.000000" : 1256,
            "30.000000" : 1272,
            "40.000000" : 1288,
            "50.000000" : 1304,
            "60.000000" : 1336,
            "70.000000" : 1384,
            "80.000000" : 1544,
            "90.000000" : 1752,
            "95.000000" : 1832,
            "99.000000" : 2256,
            "99.500000" : 2960,
            "99.900000" : 13888,
            "99.950000" : 17536,
            "99.990000" : 23936,
            "0.00" : 0,
            "0.00" : 0,
            "0.00" : 0
          }
        },
        "lat" : {
          "min" : 1076,
          "max" : 62111,
          "mean" : 1440.86,
          "stddev" : 730.83
        },
        "bw_min" : 2323,
        "bw_max" : 3025,
        "bw_agg" : 100.00,
        "bw_mean" : 2772.81,
        "bw_dev" : 164.89
      },
      "write" : {
        "io_bytes" : 0,
        "bw" : 0,
        "iops" : 0.00,
        "runtime" : 0,
        "total_ios" : 0,
        "short_ios" : 0,
        "drop_ios" : 0,
        "slat" : {
          "min" : 0,
          "max" : 0,
          "mean" : 0.00,
          "stddev" : 0.00
        },
        "clat" : {
          "min" : 0,
          "max" : 0,
          "mean" : 0.00,
          "stddev" : 0.00,
          "percentile" : {
            "1.000000" : 0,
            "5.000000" : 0,
            "10.000000" : 0,
            "20.000000" : 0,
            "30.000000" : 0,
            "40.000000" : 0,
            "50.000000" : 0,
            "60.000000" : 0,
            "70.000000" : 0,
            "80.000000" : 0,
            "90.000000" : 0,
            "95.000000" : 0,
            "99.000000" : 0,
            "99.500000" : 0,
            "99.900000" : 0,
            "99.950000" : 0,
            "99.990000" : 0,
            "0.00" : 0,
            "0.00" : 0,
            "0.00" : 0
          }
        },
        "lat" : {
          "min" : 0,
          "max" : 0,
          "mean" : 0.00,
          "stddev" : 0.00
        },
        "bw_min" : 0,
        "bw_max" : 0,
        "bw_agg" : 0.00,
        "bw_mean" : 0.00,
        "bw_dev" : 0.00
      },
      "trim" : {
        "io_bytes" : 0,
        "bw" : 0,
        "iops" : 0.00,
        "runtime" : 0,
        "total_ios" : 0,
        "short_ios" : 0,
        "drop_ios" : 0,
        "slat" : {
          "min" : 0,
          "max" : 0,
          "mean" : 0.00,
          "stddev" : 0.00
        },
        "clat" : {
          "min" : 0,
          "max" : 0,
          "mean" : 0.00,
          "stddev" : 0.00,
          "percentile" : {
            "1.000000" : 0,
            "5.000000" : 0,
            "10.000000" : 0,
            "20.000000" : 0,
            "30.000000" : 0,
            "40.000000" : 0,
            "50.000000" : 0,
            "60.000000" : 0,
            "70.000000" : 0,
            "80.000000" : 0,
            "90.000000" : 0,
            "95.000000" : 0,
            "99.000000" : 0,
            "99.500000" : 0,
            "99.900000" : 0,
            "99.950000" : 0,
            "99.990000" : 0,
            "0.00" : 0,
            "0.00" : 0,
            "0.00" : 0
          }
        },
        "lat" : {
          "min" : 0,
          "max" : 0,
          "mean" : 0.00,
          "stddev" : 0.00
        },
        "bw_min" : 0,
        "bw_max" : 0,
        "bw_agg" : 0.00,
        "bw_mean" : 0.00,
        "bw_dev" : 0.00
      },
      "usr_cpu" : 0.81,
      "sys_cpu" : 2.24,
      "ctx" : 41573,
      "majf" : 0,
      "minf" : 12,
      "iodepth_level" : {
        "1" : 100.00,
        "2" : 0.00,
        "4" : 0.00,
        "8" : 0.00,
        "16" : 0.00,
        "32" : 0.00,
        ">=64" : 0.00
      },
      "latency_us" : {
        "2" : 0.00,
        "4" : 0.00,
        "10" : 0.00,
        "20" : 0.00,
        "50" : 0.00,
        "100" : 0.00,
        "250" : 0.00,
        "500" : 0.00,
        "750" : 0.00,
        "1000" : 0.00
      },
      "latency_ms" : {
        "2" : 98.21,
        "4" : 1.51,
        "10" : 0.15,
        "20" : 0.09,
        "50" : 0.03,
        "100" : 0.01,
        "250" : 0.00,
        "500" : 0.00,
        "750" : 0.00,
        "1000" : 0.00,
        "2000" : 0.00,
        ">=2000" : 0.00
      },
      "latency_depth" : 1,
      "latency_target" : 0,
      "latency_percentile" : 100.00,
      "latency_window" : 0
    },
    {
      "jobname" : "rand-read",
      "groupid" : 1,
      "error" : 0,
      "eta" : 0,
      "elapsed" : 242,
      "read" : {
        "io_bytes" : 159192,
        "bw" : 2653,
        "iops" : 663.29,
        "runtime" : 60001,
        "total_ios" : 39798,
        "short_ios" : 0,
        "drop_ios" : 0,
        "slat" : {
          "min" : 7,
          "max" : 52,
          "mean" : 14.50,
          "stddev" : 3.10
        },
        "clat" : {
          "min" : 995,
          "max" : 38058,
          "mean" : 1489.34,
          "stddev" : 455.97,
          "percentile" : {
            "1.000000" : 1144,
            "5.000000" : 1208,
            "10.000000" : 1240,
            "20.000000" : 1272,
            "30.000000" : 1304,
            "40.000000" : 1320,
            "50.000000" : 1368,
            "60.000000" : 1432,
            "70.000000" : 1592,
            "80.000000" : 1736,
            "90.000000" : 1832,
            "95.000000" : 1928,
            "99.000000" : 2576,
            "99.500000" : 3088,
            "99.900000" : 5600,
            "99.950000" : 7200,
            "99.990000" : 17280,
            "0.00" : 0,
            "0.00" : 0,
            "0.00" : 0
          }
        },
        "lat" : {
          "min" : 1010,
          "max" : 38072,
          "mean" : 1504.36,
          "stddev" : 456.01
        },
        "bw_min" : 1980,
        "bw_max" : 2816,
        "bw_agg" : 100.00,
        "bw_mean" : 2655.40,
        "bw_dev" : 98.39
      },
      "write" : {
        "io_bytes" : 0,
        "bw" : 0,
        "iops" : 0.00,
        "runtime" : 0,
        "total_ios" : 0,
        "short_ios" : 0,
        "drop_ios" : 0,
        "slat" : {
          "min" : 0,
          "max" : 0,
          "mean" : 0.00,
          "stddev" : 0.00
        },
        "clat" : {
          "min" : 0,
          "max" : 0,
          "mean" : 0.00,
          "stddev" : 0.00,
          "percentile" : {
            "1.000000" : 0,
            "5.000000" : 0,
            "10.000000" : 0,
            "20.000000" : 0,
            "30.000000" : 0,
            "40.000000" : 0,
            "50.000000" : 0,
            "60.000000" : 0,
            "70.000000" : 0,
            "80.000000" : 0,
            "90.000000" : 0,
            "95.000000" : 0,
            "99.000000" : 0,
            "99.500000" : 0,
            "99.900000" : 0,
            "99.950000" : 0,
            "99.990000" : 0,
            "0.00" : 0,
            "0.00" : 0,
            "0.00" : 0
          }
        },
        "lat" : {
          "min" : 0,
          "max" : 0,
          "mean" : 0.00,
          "stddev" : 0.00
        },
        "bw_min" : 0,
        "bw_max" : 0,
        "bw_agg" : 0.00,
        "bw_mean" : 0.00,
        "bw_dev" : 0.00
      },
      "trim" : {
        "io_bytes" : 0,
        "bw" : 0,
        "iops" : 0.00,
        "runtime" : 0,
        "total_ios" : 0,
        "short_ios" : 0,
        "drop_ios" : 0,
        "slat" : {
          "min" : 0,
          "max" : 0,
          "mean" : 0.00,
          "stddev" : 0.00
        },
        "clat" : {
          "min" : 0,
          "max" : 0,
          "mean" : 0.00,
          "stddev" : 0.00,
          "percentile" : {
            "1.000000" : 0,
            "5.000000" : 0,
            "10.000000" : 0,
            "20.000000" : 0,
            "30.000000" : 0,
            "40.000000" : 0,
            "50.000000" : 0,
            "60.000000" : 0,
            "70.000000" : 0,
            "80.000000" : 0,
            "90.000000" : 0,
            "95.000000" : 0,
            "99.000000" : 0,
            "99.500000" : 0,
            "99.900000" : 0,
            "99.950000" : 0,
            "99.990000" : 0,
            "0.00" : 0,
            "0.00" : 0,
            "0.00" : 0
          }
        },
        "lat" : {
          "min" : 0,
          "max" : 0,
          "mean" : 0.00,
          "stddev" : 0.00
        },
        "bw_min" : 0,
        "bw_max" : 0,
        "bw_agg" : 0.00,
        "bw_mean" : 0.00,
        "bw_dev" : 0.00
      },
      "usr_cpu" : 0.53,
      "sys_cpu" : 2.39,
      "ctx" : 39800,
      "majf" : 0,
      "minf" : 11,
      "iodepth_level" : {
        "1" : 100.00,
        "2" : 0.00,
        "4" : 0.00,
        "8" : 0.00,
        "16" : 0.00,
        "32" : 0.00,
        ">=64" : 0.00
      },
      "latency_us" : {
        "2" : 0.00,
        "4" : 0.00,
        "10" : 0.00,
        "20" : 0.00,
        "50" : 0.00,
        "100" : 0.00,
        "250" : 0.00,
        "500" : 0.00,
        "750" : 0.00,
        "1000" : 0.01
      },
      "latency_ms" : {
        "2" : 96.81,
        "4" : 2.92,
        "10" : 0.24,
        "20" : 0.02,
        "50" : 0.01,
        "100" : 0.00,
        "250" : 0.00,
        "500" : 0.00,
        "750" : 0.00,
        "1000" : 0.00,
        "2000" : 0.00,
        ">=2000" : 0.00
      },
      "latency_depth" : 1,
      "latency_target" : 0,
      "latency_percentile" : 100.00,
      "latency_window" : 0
    },
    {
      "jobname" : "seq-write",
      "groupid" : 2,
      "error" : 0,
      "eta" : 0,
      "elapsed" : 242,
      "read" : {
        "io_bytes" : 0,
        "bw" : 0,
        "iops" : 0.00,
        "runtime" : 0,
        "total_ios" : 0,
        "short_ios" : 0,
        "drop_ios" : 0,
        "slat" : {
          "min" : 0,
          "max" : 0,
          "mean" : 0.00,
          "stddev" : 0.00
        },
        "clat" : {
          "min" : 0,
          "max" : 0,
          "mean" : 0.00,
          "stddev" : 0.00,
          "percentile" : {
            "1.000000" : 0,
            "5.000000" : 0,
            "10.000000" : 0,
            "20.000000" : 0,
            "30.000000" : 0,
            "40.000000" : 0,
            "50.000000" : 0,
            "60.000000" : 0,
            "70.000000" : 0,
            "80.000000" : 0,
            "90.000000" : 0,
            "95.000000" : 0,
            "99.000000" : 0,
            "99.500000" : 0,
            "99.900000" : 0,
            "99.950000" : 0,
            "99.990000" : 0,
            "0.00" : 0,
            "0.00" : 0,
            "0.00" : 0
          }
        },
        "lat" : {
          "min" : 0,
          "max" : 0,
          "mean" : 0.00,
          "stddev" : 0.00
        },
        "bw_min" : 0,
        "bw_max" : 0,
        "bw_agg" : 0.00,
        "bw_mean" : 0.00,
        "bw_dev" : 0.00
      },
      "write" : {
        "io_bytes" : 132908,
        "bw" : 2214,
        "iops" : 553.51,
        "runtime" : 60030,
        "total_ios" : 33227,
        "short_ios" : 0,
        "drop_ios" : 0,
        "slat" : {
          "min" : 10,
          "max" : 1181,
          "mean" : 17.07,
          "stddev" : 11.70
        },
        "clat" : {
          "min" : 541,
          "max" : 501675,
          "mean" : 1785.85,
          "stddev" : 12905.80,
          "percentile" : {
            "1.000000" : 1096,
            "5.000000" : 1128,
            "10.000000" : 1160,
            "20.000000" : 1176,
            "30.000000" : 1192,
            "40.000000" : 1208,
            "50.000000" : 1224,
            "60.000000" : 1240,
            "70.000000" : 1256,
            "80.000000" : 1288,
            "90.000000" : 1336,
            "95.000000" : 1416,
            "99.000000" : 2736,
            "99.500000" : 8160,
            "99.900000" : 85504,
            "99.950000" : 501760,
            "99.990000" : 501760,
            "0.00" : 0,
            "0.00" : 0,
            "0.00" : 0
          }
        },
        "lat" : {
          "min" : 1040,
          "max" : 501692,
          "mean" : 1803.49,
          "stddev" : 12906.26
        },
        "bw_min" : 180,
        "bw_max" : 3272,
        "bw_agg" : 100.00,
        "bw_mean" : 2317.92,
        "bw_dev" : 951.05
      },
      "trim" : {
        "io_bytes" : 0,
        "bw" : 0,
        "iops" : 0.00,
        "runtime" : 0,
        "total_ios" : 0,
        "short_ios" : 0,
        "drop_ios" : 0,
        "slat" : {
          "min" : 0,
          "max" : 0,
          "mean" : 0.00,
          "stddev" : 0.00
        },
        "clat" : {
          "min" : 0,
          "max" : 0,
          "mean" : 0.00,
          "stddev" : 0.00,
          "percentile" : {
            "1.000000" : 0,
            "5.000000" : 0,
            "10.000000" : 0,
            "20.000000" : 0,
            "30.000000" : 0,
            "40.000000" : 0,
            "50.000000" : 0,
            "60.000000" : 0,
            "70.000000" : 0,
            "80.000000" : 0,
            "90.000000" : 0,
            "95.000000" : 0,
            "99.000000" : 0,
            "99.500000" : 0,
            "99.900000" : 0,
            "99.950000" : 0,
            "99.990000" : 0,
            "0.00" : 0,
            "0.00" : 0,
            "0.00" : 0
          }
        },
        "lat" : {
          "min" : 0,
          "max" : 0,
          "mean" : 0.00,
          "stddev" : 0.00
        },
        "bw_min" : 0,
        "bw_max" : 0,
        "bw_agg" : 0.00,
        "bw_mean" : 0.00,
        "bw_dev" : 0.00
      },
      "usr_cpu" : 0.24,
      "sys_cpu" : 2.16,
      "ctx" : 33313,
      "majf" : 0,
      "minf" : 13,
      "iodepth_level" : {
        "1" : 100.00,
        "2" : 0.00,
        "4" : 0.00,
        "8" : 0.00,
        "16" : 0.00,
        "32" : 0.00,
        ">=64" : 0.00
      },
      "latency_us" : {
        "2" : 0.00,
        "4" : 0.00,
        "10" : 0.00,
        "20" : 0.00,
        "50" : 0.00,
        "100" : 0.00,
        "250" : 0.00,
        "500" : 0.00,
        "750" : 0.01,
        "1000" : 0.01
      },
      "latency_ms" : {
        "2" : 98.42,
        "4" : 0.82,
        "10" : 0.27,
        "20" : 0.06,
        "50" : 0.11,
        "100" : 0.23,
        "250" : 0.01,
        "500" : 0.00,
        "750" : 0.06,
        "1000" : 0.00,
        "2000" : 0.00,
        ">=2000" : 0.00
      },
      "latency_depth" : 1,
      "latency_target" : 0,
      "latency_percentile" : 100.00,
      "latency_window" : 0
    },
    {
      "jobname" : "rand-write",
      "groupid" : 3,
      "error" : 0,
      "eta" : 0,
      "elapsed" : 242,
      "read" : {
        "io_bytes" : 0,
        "bw" : 0,
        "iops" : 0.00,
        "runtime" : 0,
        "total_ios" : 0,
        "short_ios" : 0,
        "drop_ios" : 0,
        "slat" : {
          "min" : 0,
          "max" : 0,
          "mean" : 0.00,
          "stddev" : 0.00
        },
        "clat" : {
          "min" : 0,
          "max" : 0,
          "mean" : 0.00,
          "stddev" : 0.00,
          "percentile" : {
            "1.000000" : 0,
            "5.000000" : 0,
            "10.000000" : 0,
            "20.000000" : 0,
            "30.000000" : 0,
            "40.000000" : 0,
            "50.000000" : 0,
            "60.000000" : 0,
            "70.000000" : 0,
            "80.000000" : 0,
            "90.000000" : 0,
            "95.000000" : 0,
            "99.000000" : 0,
            "99.500000" : 0,
            "99.900000" : 0,
            "99.950000" : 0,
            "99.990000" : 0,
            "0.00" : 0,
            "0.00" : 0,
            "0.00" : 0
          }
        },
        "lat" : {
          "min" : 0,
          "max" : 0,
          "mean" : 0.00,
          "stddev" : 0.00
        },
        "bw_min" : 0,
        "bw_max" : 0,
        "bw_agg" : 0.00,
        "bw_mean" : 0.00,
        "bw_dev" : 0.00
      },
      "write" : {
        "io_bytes" : 31596,
        "bw" : 526,
        "iops" : 131.65,
        "runtime" : 60002,
        "total_ios" : 7899,
        "short_ios" : 0,
        "drop_ios" : 0,
        "slat" : {
          "min" : 11,
          "max" : 5724,
          "mean" : 21.75,
          "stddev" : 96.74
        },
        "clat" : {
          "min" : 3,
          "max" : 1697726,
          "mean" : 7569.51,
          "stddev" : 50646.11,
          "percentile" : {
            "1.000000" : 1144,
            "5.000000" : 1192,
            "10.000000" : 1240,
            "20.000000" : 1336,
            "30.000000" : 1432,
            "40.000000" : 1640,
            "50.000000" : 1992,
            "60.000000" : 2544,
            "70.000000" : 3280,
            "80.000000" : 4640,
            "90.000000" : 8640,
            "95.000000" : 17536,
            "99.000000" : 96768,
            "99.500000" : 101888,
            "99.900000" : 937984,
            "99.950000" : 1482752,
            "99.990000" : 1695744,
            "0.00" : 0,
            "0.00" : 0,
            "0.00" : 0
          }
        },
        "lat" : {
          "min" : 1094,
          "max" : 1697769,
          "mean" : 7591.88,
          "stddev" : 50647.86
        },
        "bw_min" : 2,
        "bw_max" : 2536,
        "bw_agg" : 100.00,
        "bw_mean" : 664.98,
        "bw_dev" : 702.49
      },
      "trim" : {
        "io_bytes" : 0,
        "bw" : 0,
        "iops" : 0.00,
        "runtime" : 0,
        "total_ios" : 0,
        "short_ios" : 0,
        "drop_ios" : 0,
        "slat" : {
          "min" : 0,
          "max" : 0,
          "mean" : 0.00,
          "stddev" : 0.00
        },
        "clat" : {
          "min" : 0,
          "max" : 0,
          "mean" : 0.00,
          "stddev" : 0.00,
          "percentile" : {
            "1.000000" : 0,
            "5.000000" : 0,
            "10.000000" : 0,
            "20.000000" : 0,
            "30.000000" : 0,
            "40.000000" : 0,
            "50.000000" : 0,
            "60.000000" : 0,
            "70.000000" : 0,
            "80.000000" : 0,
            "90.000000" : 0,
            "95.000000" : 0,
            "99.000000" : 0,
            "99.500000" : 0,
            "99.900000" : 0,
            "99.950000" : 0,
            "99.990000" : 0,
            "0.00" : 0,
            "0.00" : 0,
            "0.00" : 0
          }
        },
        "lat" : {
          "min" : 0,
          "max" : 0,
          "mean" : 0.00,
          "stddev" : 0.00
        },
        "bw_min" : 0,
        "bw_max" : 0,
        "bw_agg" : 0.00,
        "bw_mean" : 0.00,
        "bw_dev" : 0.00
      },
      "usr_cpu" : 0.14,
      "sys_cpu" : 0.40,
      "ctx" : 7957,
      "majf" : 0,
      "minf" : 11,
      "iodepth_level" : {
        "1" : 100.00,
        "2" : 0.00,
        "4" : 0.00,
        "8" : 0.00,
        "16" : 0.00,
        "32" : 0.00,
        ">=64" : 0.00
      },
      "latency_us" : {
        "2" : 0.00,
        "4" : 0.01,
        "10" : 0.00,
        "20" : 0.00,
        "50" : 0.00,
        "100" : 0.00,
        "250" : 0.00,
        "500" : 0.00,
        "750" : 0.00,
        "1000" : 0.00
      },
      "latency_ms" : {
        "2" : 50.06,
        "4" : 25.86,
        "10" : 15.57,
        "20" : 3.90,
        "50" : 2.10,
        "100" : 1.60,
        "250" : 0.68,
        "500" : 0.05,
        "750" : 0.03,
        "1000" : 0.05,
        "2000" : 0.09,
        ">=2000" : 0.00
      },
      "latency_depth" : 1,
      "latency_target" : 0,
      "latency_percentile" : 100.00,
      "latency_window" : 0
    }
  ],
  "disk_util" : [
    {
      "name" : "sda",
      "read_ios" : 81356,
      "write_ios" : 41150,
      "read_merges" : 0,
      "write_merges" : 22,
      "read_ticks" : 115656,
      "write_ticks" : 121624,
      "in_queue" : 237352,
      "util" : 97.37
    }
  ]
}
```
